tests/Interface/core/ImageContainerTest.cpp

```diff
@@ -24,6 +24,45 @@
 #include <buddy/Core/Container.h>
 #include <buddy/DIP/ImageContainer.h>
 
+bool compare_flt(float a,float b)
+{
+	if(a==b) return true;
+        float eps = a-b;
+        if(eps<0)eps=-eps;
+        if(eps<FLT_EPSILON) return true;
+        return false;
+}
+template<typename T,size_t N>
+bool testcvdp(cv::Mat testImgcv,Img<T,N> testImg,bool norm = false)
+{
+	int cvn = testImgcv.dims;
+	if(cvn!=N) return false;
+	for(int i=0;i<N;i++)
+	{
+		if(testImgcv.size[i]!=testImg.getSizes()[i]) return false;
+	}
+	T *data= testImg.getData();
+	if(N==2)
+	{
+		int k=0;
+		for(size_t i=0;i<testImg.getSizes()[0];i++)
+		{
+			for(size_t j=0;j<testImg.getSizes()[1];j++)
+			{
+				if(!norm)
+				{
+					if(!compare_flt(data[k],(T)testImgcv.at<uchar>(i,j))) return false;
+				}
+				else
+				{
+					if(!compare_flt(data[k],(T)testImgcv.at<T>(i,j)))return false;
```

Comment:
Both of these branches (if and else block) can be simplified using a ternary operator.